en/api-guide.md

```diff
@@ -156,3 +156,204 @@ Content-Type: application/json
+        "url": "http://toast.com",
+        "httpMethod": "GET",
+        "validation": {
+            "textValidation": {
+                "textValidationType": "JSON",
+                "textValidationInfos": [
+                    {
+                        "operator": "EQ",
+                        "expression": "$.isSuccess",
+                        "operand": "true"
+                    }
+                ]
+            },
+            "responseCodes": [
+                "200",
+                "201"
+            ]
+        },
+        "browserOption": {
+            "OPT_LOCALE": "ko"
+        },
+        "ip": "toast.com",
+        "scenarioType": "API",
+        "scenarioName": "API test",
+        "description": "API test",
+        "monitoringRegion": [
+            "KOR"
+        ],
+        "amendedTime": "2020-09-03T08:49:01.197+0000",
+        "monitoringCron": "7 * * * * ? *",
+        "status": "temporary",
+        "errorLimitCount": 0
+    }
+}
+```
+
+값 | 타입 | 설명
+--- | --- | ---
+header.isSuccessful | Boolean | 성공 여부
+header.resultCode | Integer | 실패 코드(0은 정상)
+header.resultMessage | String | 실패 메시지
+body.scenarioId | String | 시나리오 ID
+body.url  |  String  |  모니터링할 API의 URL
+headers  |  Map&lt;String, String&gt;  |  API를 보낼 때 사용할 헤더값
+body.httpMethod  |  String  |  API의 httpMethod
+body.requestBody  |  String  |  API의 requestBody
+body.browserOption  |  Map&lt;String, String&gt;  |  
+[body.validation](#validation2)  |  Object  |  API의 검증 정보
+body.scenarioType  |  String  |  시나리오 타입
+body.scenarioName  |  String  |  시나리오 이름
+body.description  |  String  |  시나리오 설명
+body.monitoringRegion  |  Set&lt;String&gt;  |  시나리오를 모니터링할 지역
+body.monitoringInterval  |  Integer  |  모니터링 간격(초)
+body.monitoringCron  |  String  |  모니터링 간격(Cron 표현식)
+body.errorLimitCount  |  Integer  |  연속 오류 허용 횟수
+body.registeredTime | String | 등록 시각(yyyy-MM-dd'T'HH:mm:ss.SSSz)
+body.amendedTime | String | 수정 시각(yyyy-MM-dd'T'HH:mm:ss.SSSz)
+body.status | String | 시나리오의 현재 상태
+
+<div id='validation2'></div>
+- validation
+
+값  |  타입  | 설명
+--- | --- | ---
+[body.validation.textValidation](#textValidation2)  |  Object  |  문자열 검증 정보
+body.validation.timeout  |  Integer  |  타임아웃 임곗값
+body.validation.responseCodes  | Set&lt;String&gt;  |  허용된 responseCode
+body.validation.avoidingValidationText  |  String  |  body에 포함된 경우 전파를 제외할 문자열
+
+<div id='textValidation2'></div>
+- textValidation
+
+필드명(경로명)  |  타입  |  설명
+--- | --- | ---
+textValidationType  |  String  |  문자열을 검증할 때 기반이 되는 body 타입
+[body.validation.textValidation.textValidationInfos](#textValidationInfo2)  |  List&lt;Object&gt;  |  문자열 검증 정보
+
+<div id='textValidationInfo2'></div>
+- textValidationInfo
+
+값  |  타입  |  설명
+--- | --- | ---
+body.validation.textValidation.textValidationInfo.operator  |  String  |  문자열 연산자
+body.validation.textValidation.textValidationInfo.expression  |  String  |  검증이 필요한 문자열
+body.validation.textValidation.textValidationInfo.operand  |  String  |  기댓값
```

Comment:
## Scenario Modification

### Data transfer
- Send data required for scenario modification request to the service monitoring server.

[URL]

```http
POST /open-api/v1.0/appkey/{appKey}/scenarios/{scenarioId}
Content-Type: application/json
```

[Path Variables]

| Value |Type | Necessity |Description |
|---|---|---|---|
| appKey | String | Required | Service Appkey (Viewable in **Manage Service** tab) |
| scenarioId | String | Required | Scenario ID (Viewable in **Edit Scenario** window) |

[Request Header]

 Header Name | Header Value 
 --- | ---
 TOAST_PRODUCT_APPKEY | Service Monitoring: In the **Manage Service** menu, click **URL & Appkey** on the upper right side to check this Appkey. 

[Request Body]
```json
{
   "url":"http://toast.com",
   "httpMethod":"GET",
   "validation":{
      "textValidation":{
         "textValidationType":"JSON",
         "textValidationInfos":[
            {
               "expression":"$.isSuccess",
               "operator":"EQ",
               "operand":"true"
            }
         ]
      },
      "responseCodes":[
         "200",
         "201"
      ]
   },
   "browserOption":{
      "OPT_LOCALE":"ko"
   },
   "ip":"toast.com",
   "scenarioType":"API",
   "scenarioName":"API test",
   "description":"API test",
   "monitoringRegion":[
      "KOR"
   ],
   "monitoringInterval":30,
   "errorLimitCount":0
}
```

- Request Body

Value | Type | Corresponding scenarioType | Assignable Value | Necessity | Default | Description
---|---|---|---|---|---|---
url | String | API | url starting with http or https | Y |  | The URL of API to monitor 
headers | Map&lt;String, String&gt; | API |  | N |  | Header value to use to send the API
httpMethod | String | API | GET, POST, DELETE, PUT | Y |  | httpMethod of API
requestBody | String | API |  | N |  | requestBody of API
browserOption | Map&lt;String, String&gt; | API | {"OPT_LOCALE" : "kr"} | Y | {"OPT_LOCALE" : "kr"} | 
[validation](#validation1) | Object | API |  | Y |  | Validation info of API
scenarioType | String | API | API | Y |  | Scenario type
scenarioName | String | API |  | Y |  | Scenario name
description | String | API |  | Y |  | Scenario description
monitoringRegion | Set&lt;String&gt; | API | KOR, US | Y | KOR | The region to monitor a scenario
monitoringInterval | Integer | API |  | N (monitoringCron required if this is not used) |  | Monitoring intervals (sec)
monitoringCron | String | API | 5-digit Cron expression | N (monitoringInterval required if this is not used) |  | Monitoring intervals (Cron expression)
errorLimitCount | Integer | API | 0 or higher integer | Y | 0 | Number of repeat errors allowed

<div id='validation1'></div>

- validation

Value | Type | Corresponding scenarioType | Assignable Value | Necessity | Default | Description
---|---|---|---|---|---|---
[validation.textValidation](#textValidation1) | Object | API |  | N |  | String validation info
validation.timeout | Integer | API | 0 or higher integer (in ms) | N |  | Timeout threshold
validation.responseCodes | Set&lt;String&gt; | API | HTTP response code | N |  | Allowed response code
validation.avoidingValidationText | String | API |  | N |  | String to exclude from propagation, if included in the body

<div id='textValidation1'></div>
- textValidation

Value | Type | Corresponding scenarioType | Assignable Value | Necessity | Default | Description
---|---|---|---|---|---|---
validation.textValidation.textValidationType | String | API | JSON, HTML, and XML | N |  | Base body type to validate a string
[validation.textValidation.textValidationInfos](#textValidationInfo1) | List&lt;Object&gt; | API | | N |  | String validation info

<div id='textValidationInfo1'></div>
- textValidationInfo

Value | Type | Corresponding scenarioType | Assignable Value | Necessity | Default | Description
---|---|---|---|---|---|---
validation.textValidation.textValidationInfo.operator | String | API | CONTAINS, NOT_CONTAINS, EQ, NE, GT, GTE, LT, LTE | Y |  | String operator
validation.textValidation.textValidationInfo.expression | String | API |  | Y |  | String that requires validation
validation.textValidation.textValidationInfo.operand | String | API |  | Y(N) |  | Expected Value

#### Response
```json
{
    "header": {
        "isSuccessful": true,
        "resultCode": 0,
        "resultMessage": "SUCCESS"
    },
    "body": {
        "scenarioId": "0c96cff0-edc2-11ea-9760-8d94f461e6d4",
        "url": "http://toast.com",
        "httpMethod": "GET",
        "validation": {
            "textValidation": {
                "textValidationType": "JSON",
                "textValidationInfos": [
                    {
                        "operator": "EQ",
                        "expression": "$.isSuccess",
                        "operand": "true"
                    }
                ]
            },
            "responseCodes": [
                "200",
                "201"
            ]
        },
        "browserOption": {
            "OPT_LOCALE": "ko"
        },
        "ip": "toast.com",
        "scenarioType": "API",
        "scenarioName": "API test",
        "description": "API test",
        "monitoringRegion": [
            "KOR"
        ],
        "amendedTime": "2020-09-03T08:49:01.197+0000",
        "monitoringCron": "7 * * * * ? *",
        "status": "temporary",
        "errorLimitCount": 0
    }
}
```

Value | Type | Description
--- | --- | ---
header.isSuccessful | Boolean | Success
header.resultCode | Integer | Failure Code (0: Normal)
header.resultMessage | String | Failure message
body.scenarioId | String | Scenario ID
body.url  |  String  | The URL of API to monitor 
headers  |  Map&lt;String, String&gt;  |  Header value to use to send the API
body.httpMethod  |  String  |  httpMethod of API
body.requestBody  |  String  |  requestBody of API
body.browserOption  |  Map&lt;String, String&gt;  |  
[body.validation](#validation2)  |  Object  |  Validation info of API
body.scenarioType  |  String  |  Scenario type
body.scenarioName  |  String  |  Scenario name
body.description  |  String  |  Scenario description
body.monitoringRegion  |  Set&lt;String&gt;  |  The region to monitor a scenario
body.monitoringInterval  |  Integer  |  Monitoring intervals (sec)
body.monitoringCron  |  String  |  Monitoring intervals (Cron expression)
body.errorLimitCount  |  Integer  |  Number of repeat errors allowed
body.registeredTime | String | Registered time (yyyy-MM-dd'T'HH:mm:ss.SSSz)
body.amendedTime | String | Modified time (yyyy-MM-dd'T'HH:mm:ss.SSSz)
body.status | String | Scenario's current status

<div id='validation2'></div>
- validation

Value  |  Type  | Description
--- | --- | ---
[body.validation.textValidation](#textValidation2)  |  Object  |  String validation info
body.validation.timeout  |  Integer  |  Timeout threshold
body.validation.responseCodes  | Set&lt;String&gt;  |  Allowed response code
body.validation.avoidingValidationText  |  String  |  String to exclude from propagation, if included in the body

<div id='textValidation2'></div>
- textValidation

Field name (directory name)  |  Type  |  Description
--- | --- | ---
textValidationType  |  String  |  Base body type to validate a string
[body.validation.textValidation.textValidationInfos](#textValidationInfo2)  |  List&lt;Object&gt;  |  String validation info

<div id='textValidationInfo2'></div>
- textValidationInfo

Value  |  Type  |  Description
--- | --- | ---
body.validation.textValidation.textValidationInfo.operator  |  String  |  String operator
body.validation.textValidation.textValidationInfo.expression  |  String  |  String that requires validation
body.validation.textValidation.textValidationInfo.operand  |  String  |  Expected Value